app/scripts/metamask-controller.js

```diff
@@ -7810,7 +7575,12 @@ export default class MetamaskController extends EventEmitter {
       getStateUI: this._getMetaMaskState.bind(this),
       getTransactionMetricsRequest:
         this.getTransactionMetricsRequest.bind(this),
+      offscreenPromise: this.offscreenPromise,
       persistedState: initState,
+      removeAllConnections: this.removeAllConnections.bind(this),
```

Comment:
Wondering if `initRequest` will end up bloated - any ideas how we can reduce the additions here? Maybe in a future PR as well?

Reply:
These properties are all very tied to the internal state of `MetaMaskController`. Some things could be moved to (new) controllers in the future, but that feels out of scope of this PR and out of scope for the Snaps team in general.